File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> reverseThree; 
    //n(10)->3진법 역순 저장
    while(n){
        reverseThree.push_back(n%3); 
        n = n/3;
    }
    
    //다시 10진법으로 변환 
    int multiplier = 1; 
    for(int i= reverseThree.size()-1; i>=0; i--){
        answer += reverseThree[i]*multiplier;
        multiplier *= 3; 
    }
    return answer;
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        //10진법 수 -> 3진법으로 만들기
        int moc = 0; 
        int rem = 0;
        String s = ""; 
        
        while(true){
            moc = n/3; 
            rem = n%3; 
            s += (char) (rem+'0');
            if(moc==0) break;
            n = moc; 
        }
                
        //앞뒤 뒤집기: 이미 뒤집어져있음
                
        //뒤집은 3진법 수를 10진법으로 표현하기 
        for(int i=0; i<s.length(); i++){
            answer += Math.pow(3, (s.length()-i-1))*(s.charAt(i)-'0');
        }
        return answer;
    }
}